### PR TITLE
Prevent TS compiler from running in parallel

### PR DESCRIPTION
**Pull request by ry**
Fixes 015_duplicate_parallel_import

ref https://github.com/denoland/deno/issues/3876

### CODE REVIEW

File: cli/compilers/json.rs [hunk 1]
```diff
@@ -1,32 +1,30 @@
...
 impl JsonCompiler {
-  pub fn compile_async(
+  pub async fn compile_async(
     &self,
     source_file: &SourceFile,
-  ) -> Pin<Box<CompiledModuleFuture>> {
+  ) -> Result<CompiledModule, ErrBox> {
     let maybe_json_value: serde_json::Result<serde_json::Value> =
-      serde_json::from_str(&str::from_utf8(&source_file.source_code).unwrap());
+      serde_json::from_str(
```
Comment (piscisaureus) [inline cli/compilers/json.rs]: While you're at it, maybe remove the str::from_utf8 part and use serde_json::from_slice ?

File: cli/compilers/ts.rs [hunk 2]
```diff
@@ -216,6 +217,22 @@ pub struct TsCompiler {
...
+impl Deref for TsCompiler {
+  type Target = TsCompilerInner;
+  fn deref(&self) -> &Self::Target {
+    &self.0
+  }
+}
+
+impl DerefMut for TsCompiler {
+  fn deref_mut(&mut self) -> &mut Self::Target {
+    Arc::get_mut(&mut self.0).unwrap()
```
Comment (piscisaureus) [inline cli/compilers/ts.rs]: Are you sure you want to use Arc::get_mut() here? Note that the latter will fail if the Arc has > 1 references to it.
I suspect you really just want to use `Option<TSCompiler>` to store the thing and `as_mut()` to deref it.

Comment (ry) [reply cli/compilers/ts.rs]: removed

File: cli/compilers/json.rs (hunk 1 above)
Comment (ry) [reply cli/compilers/json.rs]: done

**Pull request merged**